### DISCUSSION TITLE
How to verify that a control meets the input requirements

### DISCUSSION THREAD

**Discussion by hui3215** [AvaloniaUI/Avalonia#8687] [Q&A]
How does the TextBox control validate input?

**Comment by maxkatz6** [comment]
There are some multiple approaches https://github.com/AvaloniaUI/Avalonia/tree/master/samples/BindingDemo/ViewModels

Data annotations attributes, throwing an exception or INotifyDataErrorInfo implementation.
Or any library that works on top of them, like ReactiveUI.Validation.
